ClangSharpPInvokeGenerator/CXCursorExtensions.cs

```diff
@@ -0,0 +1,229 @@
+            Debug.Assert(cursor.Kind == CXCursorKind.CXCursor_EnumDecl);
+            return cursor.EnumDecl_IntegerType.GetName(cursor);
+        }
+
+        public static string GetEnumDeclName(this CXCursor cursor)
+        {
+            Debug.Assert(cursor.Kind == CXCursorKind.CXCursor_EnumDecl);
+
+            var name = cursor.Spelling.ToString();
+
+            if (string.IsNullOrWhiteSpace(name))
+            {
+                name = cursor.Type.GetName(cursor);
+            }
+
+            Debug.Assert(!string.IsNullOrWhiteSpace(name));
+            return EscapeName(name);
+        }
+
+        public static string GetFieldDeclName(this CXCursor cursor)
+        {
+            Debug.Assert(cursor.Kind == CXCursorKind.CXCursor_FieldDecl);
+
+            var name = cursor.Spelling.ToString();
+            Debug.Assert(!string.IsNullOrWhiteSpace(name));
+            return EscapeName(name);
+        }
+
+        public static string GetFieldDeclTypeName(this CXCursor cursor)
+        {
+            Debug.Assert(cursor.Kind == CXCursorKind.CXCursor_FieldDecl);
+            return cursor.Type.GetName(cursor);
+        }
+
+        public static string GetFunctionDeclName(this CXCursor cursor)
+        {
+            Debug.Assert(cursor.Kind == CXCursorKind.CXCursor_FunctionDecl);
+
+            var name = cursor.Spelling.ToString();
+            Debug.Assert(!string.IsNullOrWhiteSpace(name));
+            return EscapeName(name);
+        }
+
+        public static string GetParmDeclName(this CXCursor cursor, int index)
+        {
+            Debug.Assert(cursor.Kind == CXCursorKind.CXCursor_ParmDecl);
+
+            var name = cursor.Spelling.ToString();
+
+            if (string.IsNullOrWhiteSpace(name))
+            {
+                name = $"param{index}";
+            }
+
+            Debug.Assert(!string.IsNullOrWhiteSpace(name));
+            return EscapeName(name);
+        }
+
+        public static string GetParmDeclTypeName(this CXCursor cursor)
+        {
+            Debug.Assert(cursor.Kind == CXCursorKind.CXCursor_ParmDecl);
+
+            var name = cursor.Type.GetName(cursor);
+            Debug.Assert(!string.IsNullOrWhiteSpace(name));
+            return name;
+        }
+
+        public static string GetStructDeclName(this CXCursor cursor)
+        {
+            Debug.Assert(cursor.Kind == CXCursorKind.CXCursor_StructDecl);
+
+            var name = cursor.Spelling.ToString();
+
+            if (string.IsNullOrWhiteSpace(name))
+            {
+                name = cursor.Type.GetName(cursor);
+            }
+
+            Debug.Assert(!string.IsNullOrWhiteSpace(name));
+            return EscapeName(name);
+        }
+
+        public static string GetTypedefDeclName(this CXCursor cursor)
+        {
+            Debug.Assert(cursor.Kind == CXCursorKind.CXCursor_TypedefDecl);
+
+            var name = cursor.Spelling.ToString();
+
+            switch (name)
+            {
+                case "size_t":
```

Comment:
Hopefully a better can be solved to handle this and other similar types... Potentially could check for both regular and `-m32` mode and if it differs chooise `IntPtr`, that way it is more general....